frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Link } from 'react-router-dom';
+import { Tooltip, Button } from '@patternfly/react-core';
+import { StatusBox, LoadingBox } from '@console/internal/components/utils';
+import Topology from '../Topology';
+import { TopologyPage, RenderTopology } from '../TopologyPage';
+import NamespacedPage from '../../NamespacedPage';
+import { topologyData } from './topology-test-data';
+
+type TopologyPageProps = React.ComponentProps<typeof TopologyPage>;
+type RenderTopologyProps = React.ComponentProps<typeof RenderTopology>;
+
+let topologyProps: TopologyPageProps;
+let renderTopologyProps: RenderTopologyProps;
+
+describe('Topology page tests', () => {
```

Comment:
I think your tests are a little too specific. Tying to the static implementation is just asking to couple our tests to the rendering rather than effectively testing the possible outputs of the component.

I think you have really 2 tests to write:

* Did it render Topology by default
* Did it render Workload List when on the specific route

There isn't a lot of other things to test here other than the icon flipping and the content changing. 

We shouldn't be looking to test every facet of a page, otherwise we'll fail tests just by doing minor corrections and tests will hurt more than they help.